src/astronomicalObject/AstronomicalObject.ts

```diff
@@ -1,42 +1,44 @@
-import TimeOfInterest from '../time/TimeOfInterest';
 import IAstronomicalObject from './interfaces/IAstronomicalObject';
-import {createTimeOfInterest} from '../time';
+import { LIGHT_SPEED_KM_PER_SEC } from '../constants/lightSpeed';
 import {
     EclipticSphericalCoordinates,
     EquatorialSphericalCoordinates,
     LocalHorizontalCoordinates,
     RectangularCoordinates
-} from '../coordinates/coordinateTypes';
-import {getConjunctionInLongitude, getConjunctionInRightAscension} from '../utils/conjunctionCalc';
+    } from '../coordinates/coordinateTypes';
+import { Location } from '../earth/LocationTypes';
+import { Conjunction } from '../planets/planetTypes';
+import { createTimeOfInterest } from '../time';
+import TimeOfInterest from '../time/TimeOfInterest';
+import { correctEffectOfRefraction } from '../utils/apparentCoordinateCalc';
+import { getConjunctionInLongitude, getConjunctionInRightAscension } from '../utils/conjunctionCalc';
 import {
     eclipticSpherical2equatorialSpherical,
     equatorialSpherical2topocentricHorizontal,
     equatorialSpherical2topocentricSpherical,
     spherical2rectangular
-} from '../utils/coordinateCalc';
-import {au2km} from '../utils/distanceCalc';
-import {LIGHT_SPEED_KM_PER_SEC} from '../constants/lightSpeed';
-import {Location} from '../earth/LocationTypes';
-import {correctEffectOfRefraction} from '../utils/apparentCoordinateCalc';
-import {Conjunction} from '../planets/planetTypes';
+    } from '../utils/coordinateCalc';
+import { au2km } from '../utils/distanceCalc';
 
 export default abstract class AstronomicalObject implements IAstronomicalObject {
-    protected name = 'astronomical object';
 
-    protected jd: number = 0.0;
-    protected jd0: number = 0.0;
-    protected T: number = 0.0;
-    protected t: number = 0.0;
+    protected readonly jd: number;
```

Comment:
Commit for `readonly`: https://github.com/andrmoel/astronomy-bundle-js/commit/2737a3a7b0b38deb6c4051b3466554e7039f4243